### ISSUE TITLE
Setting a user defined profile in ppd.conf crashes ppd service

### ISSUE THREAD

**Issue by ponychicken** [redhat-performance/tuned#667]
[tuned/ppd/controller.py](https://github.com/redhat-performance/tuned/commit/dd0f0fb317c7147104c0f080c86c90b3287cbcd2#r145015460)

dd0f0fb317c7147104c0f080c86c90b3287cbcd2 Introduced a bug in the init.  If switch_profile does not return early, it will look up self_on_battery, which is just set with the next call. For some reason this only becomes a problem when using user defined profiles, it works fine if using a system profile

**Comment by ponychicken** [comment]
```
  File "/usr/local/lib/python3.11/dist-packages/tuned/ppd/controller.py", line 106, in __init__
    self.initialize()
  File "/usr/local/lib/python3.11/dist-packages/tuned/ppd/controller.py", line 141, in initialize
    self.switch_profile(self._base_profile)
  File "/usr/local/lib/python3.11/dist-packages/tuned/ppd/controller.py", line 166, in switch_profile
    tuned_profile = self._config.ppd_to_tuned_battery[profile] if self._on_battery else self._config.ppd_to_tuned[profile]
                                                                  ^^^^^^^^^^^^^^^^
AttributeError: 'Controller' object has no attribute '_on_battery'
```

**Comment by zacikpa** [comment]
Closing this, as it is fixed in #672.
